x-pack/libbeat/management/fleet/manager.go

```diff
@@ -122,38 +126,52 @@ func (cm *Manager) CheckRawConfig(cfg *common.Config) error {
 	return nil
 }
 
+// UpdateStatus updates the manager with the current status for the beat.
+func (cm *Manager) UpdateStatus(status management.Status, msg string) {
+	cm.lock.Lock()
+	if cm.status != status || cm.msg != msg {
+		cm.status = status
+		cm.msg = msg
+		cm.lock.Unlock()
```

Comment:
we should test this with concurrent request, what i fear may occur is that wit many concurrent changes of status
we will end up with different `internal state` vs `reported state`
do you see some pros/cons of doing reporting using a channel or having full flow lock?

Reply:
We can also try to 'forget' outdated status updates if we get new ones while we are waiting. This could be implemented using `sync.Cond`.

Reply:
My only reason to track the state at this level is so it can be logged when it is changed. We could remove the internal tracking and locking in the Fleet Manager if we just log every time `UpdateStatus` is called. Then the locking inside of `client` will handle updating the correct state to the manager.

Or we could place the call to `client` inside of the lock to ensure that its always the same between the 2.

Reply:
i think client options is also ok. i think logging inconsistent states will cause us more trouble then help us during some sdh triaging

Reply:
I went with moving the logging and client into the lock.